book/src/appendices/b_migration_notes/cgmath_to_nalgebra.md

```diff
@@ -0,0 +1,228 @@
+# CgMath to Nalgebra
+
+## How To Use
+
+This cheat sheet is split up into the following sections:
+
+* **Type Changes:** Previously you used `this::Type`, now you use `another::Thing`
+* **Logic Changes:** Previously you had `object.method(args)`, now you use `object.other(stuff)`
+
+Most changes will have accompanying explanations and code examples on how to switch.
+
+This document is by no means exhaustive, so if there is something missing, or if you can clarify any changes, please correct this!
+
+The text is designed to be searchable, so if you are looking for a specific type or method, please use Ctrl + F: `TypeName`. If you cannot find it in the document, likely we missed it during writing. Please send us a pull request!
+
+## Type Changes
+
+Many types retain the same type name, just under the `nalgebra` namespace:
+
+```patch
+-use amethyst::core::cgmath::{Vector2, Vector3, Matrix4};
++use amethyst::core::nalgebra::{Vector2, Vector3, Matrix4};
+```
+
+We will not list the names of every type with the same simple name, but will try to list the changes for types whose simple names are different:
+
+```patch
+-cgmath::Ortho
++nalgebra::Orthographic3
+
+-cgmath::PerspectiveFov
++nalgebra::Perspective3
+```
+
+## Logic Changes
+
+* `cgmath` to `nalgebra` functions:
+
+    ```patch
+    -Vector3::unit_z()
+    +Vector3::z()
+
+    -matrix4.z.truncate()
+    +matrix4.column(2).xyz().into()
+
+    -matrix4.transform_point(origin)
+    +matrix4.transform_point(&origin)
+    ```
+
+* `amethyst::renderer::Transform`
+
+    - Stores translation and rotation in an `iso: Isometry` field, and scaling in a `Vector3`.
```

Comment:
I'm not sure this needs to be mentioned as much as we should mention that you no longer can directly access the fields of the `Transform`.

Reply:
I think mentioning both is fine. If you want to, we can say that's how it works (and not accessed).

Reply:
But why would a user need to know how it's stored? As long as it's exposed decently with accessor and it's performant, the user should not have to care.